cooperate.sql

```diff
@@ -0,0 +1,61 @@
+DROP SCHEMA public CASCADE;
+CREATE SCHEMA public;
+
+DROP DATABASE cooperate;
+CREATE DATABASE cooperate;
+
+\c cooperate;
+--This table contains login info for user as well as their karma
+
+CREATE TABLE Users (
+  id varchar PRIMARY KEY NOT NULL, --username
+  email_address varchar NOT NULL, --must be a cooper email address
+  hashed_password integer NOT NULL, --salted hashed password used
+  karma int NOT NULL DEFAULT 0, --net number of likes that a user gets
+  created_at timestamp NOT NULL
+);
+
+--This table stores course information and the aggregate rating
+CREATE TABLE Course (
+  id varchar PRIMARY KEY NOT NULL,
+  descrip text NOT NULL, --description of course
+  rating float --weighted average rating of course
+);
+
+--This table stores prof information and aggregate rating
+CREATE TABLE Professor (
+  id varchar PRIMARY KEY,
+  descrip text, --about me for professor
+  rating float --weighted average rating for professor
+);
+
+--This table contains the reviews posted by our users
+CREATE SEQUENCE review_counter start with 10000;
+CREATE TABLE Reviews (
+  review_id int NOT NULL DEFAULT nextval('review_counter'),
+  course_id varchar NOT NULL, --primary key for course being reviewed
+  prof_id varchar NOT NULL, --primary key for professor being reviewed
+  user_id varchar NOT NULL, -- primary key of user posting review
+  review text NOT NULL, --the review of the course
+  course_rating float NOT NULL, --course rating 0 to 1 in increments of .2
+  prof_rating float NOT NULL, --prof rating 0 to 1 in increments of .2
+  net_likes integer NOT NULL DEFAULT 0, --net number of likes given by users
+  hyperlink varchar, --hyperlink to the google drive folder in which any attached documents for review are stored
+  created_at timestamp NOT NULL,
+
+  PRIMARY KEY(review_id),
+  FOREIGN KEY (course_id) REFERENCES Course,
+  FOREIGN KEY (prof_id) REFERENCES Professor,
+  FOREIGN KEY (user_id) REFERENCES Users
+);
+
+--This table stores the likes and dislikes made by a user for a review
+CREATE TABLE Likes (
+  user_id varchar NOT NULL,
```

Comment:
I'm not sure if you need to mention these twice if they're being referenced with the foreign key already. Just saying out of caution

Reply:
I agree that it isn't really elegant to have to create a column in one line and then declare it as a foreign key in a separate line. Unfortunately, it looks like there isn't any good way to do it in one line in postgres which is pretty strange: https://dba.stackexchange.com/questions/132029/how-to-add-a-column-with-a-foreign-key-constraint-to-a-table-that-already-exists